server/sonar-server/src/main/java/org/sonar/server/startup/RenameIssueWidgets.java

```diff
@@ -0,0 +1,161 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.startup;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import org.picocontainer.Startable;
+import org.sonar.api.utils.System2;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.core.dashboard.WidgetDto;
+import org.sonar.core.dashboard.WidgetPropertyDto;
+import org.sonar.core.issue.db.IssueFilterDto;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.core.persistence.MyBatis;
+import org.sonar.core.template.LoadedTemplateDto;
+import org.sonar.server.db.DbClient;
+import org.sonar.server.issue.filter.RegisterIssueFilters;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class RenameIssueWidgets implements Startable {
+
+  private final static String TASK_KEY = "RenameIssueWidgets";
+
+  private static final String WIDGET_FALSE_POSITIVES = "false_positive_reviews";
+  private static final String WIDGET_MY_UNRESOLVED = "my_reviews";
+  private static final String WIDGET_UNRESOLVED_BY_DEVELOPER = "reviews_per_developer";
+  private static final String WIDGET_UNRESOLVED_BY_STATUS = "unresolved_issues_statuses";
+
+  private static final Set<String> WIDGETS_TO_REPLACE = ImmutableSet.of(
+    WIDGET_FALSE_POSITIVES,
+    WIDGET_MY_UNRESOLVED,
+    WIDGET_UNRESOLVED_BY_DEVELOPER,
+    WIDGET_UNRESOLVED_BY_STATUS
+  );
+
+  private static final String WIDGET_PROJECT_ISSUE_FILTER = "project_issue_filter";
+
+  private static final String FILTER_PROPERTY = "filter";
+  private static final String DISTRIBUTION_AXIS_PROPERTY = "distributionAxis";
+
+  private final DbClient dbClient;
+
+  private final System2 system;
+
+  public RenameIssueWidgets(DbClient dbClient, System2 system, RegisterIssueFilters startupDependency) {
+    this.dbClient = dbClient;
+    this.system = system;
+    // RegisterIssueFilters must be run before this task, to be able to reference issue filters in widget properties
+  }
+
+  @Override
+  public void start() {
+    DbSession session = dbClient.openSession(false);
+
+    try {
+      if (dbClient.loadedTemplateDao().countByTypeAndKey(LoadedTemplateDto.ONE_SHOT_TASK_TYPE, TASK_KEY) != 0) {
+        // Already done
+        return;
+      }
+
+      Map<String, IssueFilterDto> filterByWidgetKey = loadRequiredIssueFilters();
+
+      Map<String, String> distributionAxisByWidgetKey = ImmutableMap.of(
+        WIDGET_FALSE_POSITIVES, "resolutions",
+        WIDGET_MY_UNRESOLVED, "severities",
+        WIDGET_UNRESOLVED_BY_DEVELOPER, "assignees",
+        WIDGET_UNRESOLVED_BY_STATUS, "statuses"
+      );
```

Comment:
Far better ! And in fact you could replace WIDGETS_TO_REPLACE set by this map, and use WIDGETS_TO_REPLACE.keys() when you only need the widget keys.